### DISCUSSION TITLE
Working with properties while writing scripts: Can't find a property

### DISCUSSION THREAD

**Discussion by spacemanmartin** [Stellarium/stellarium#3086] [Q&A] [answered]
Would someone be able to help with a scripting question related to the "Moon Age", which is apparently one of the "Additional Information" attributes you can get for the moon?

I would simply like to output the "Moon age" with a script in Stellarium, but I don't seem to be able to find the associated property.

For example, if I want to use a loop to get the moon's phase as a function of JD, I can write:
objectData=core.getSelectedObjectInfo();
core.output( core.getJDay() + "," + objectData["phase"]);

I've tried to output all the available properties and there just doesn't seem to be one for Moon age, e.g. :
core.mapToString(objectData);

Where am I going wrong? Many thanks in advance for any hints / solutions!

**Comment by alex-w** [comment]
Moon age and name of phase data are not available through scripting engine.

**Comment by gzotti** [comment, marked as answer]
Hmm, this should then be added to the getInfoMap() call...
I have not checked: is the Moon age still just the elongation-derived age, or actual time since last conjunction?

**Reply by alex-w** [reply]
> Hmm, this should then be added to the getInfoMap() call... I have not checked: is the Moon age still just the elongation-derived age, or actual time since last conjunction?

https://github.com/Stellarium/stellarium/blob/b4a5caceeba9f758e63f0f62c8f9c534ba1c21ed/src/core/modules/Planet.cpp#L1224

**Reply by gzotti** [reply]
OK, angle-derived age. Most users expect "time since conjunction", which is clearly more difficult. Possibly date of last conjunction should be explicitly computed and cached for age computing.

**Reply by spacemanmartin** [reply]
Wow! Thanks for the amazing response time!!!

That's too bad about this parameter not being available yet. 
However, the calculations of lines 1241 and 1265 from the code linked above are interesting:

deltaLong = StelUtils::fmodpos((lambdaMoon-lambdaSun)*M_180_PI, 360.);
age = deltaLong*29.530588853/360.;

What does "lambdaMoon" and "lambdaSun" represent? Are both of these parameters accessible as properties somewhere, by any chance, such that I could try to actually build these calculations into my Stellarium script itself as a workaround? The moon seems to have an accessible property "elongation", but the sun doesn't. 

If this is not currently feasible, is there any other approach I could try to script an output report of some kind of metric of the current point in the lunar cycle or Moon phase as a function of JD reported? Can I use only illumination to calculate this somehow, even though there's an ambiguity of waxing/waning for each possible value between 0% and 100%?

Thanks again.

**Reply by gzotti** [reply]
lambda is the symbol for ecliptical longitude.
elongation is (angular) distance from the sun. The sun cannot have one...
Of course you can re-implement these lines in JS if you need them next Friday. However, there are about 2 weeks to 23.1. Not having these data in InfoMap looks like a tiny omission to me that can be fixed by then. Fixing "real" age  (time since conjunction) seems more difficult.

**Comment by spacemanmartin** [comment]
Thank you again. I certainly have no urgent need for this before next Friday, haha! I didn't realize a new version of the software would be coming out so soon (how lucky!) I am happy to wait to re-install the new version when it's released, which will enable the Moon Age to be queried in the normal way. Thanks for making this software even more amazing in the future!

**Reply by gzotti** [reply]
Solved in #3088
